lib/kamal/cli/app/ssl_certificates.rb

```diff
@@ -0,0 +1,28 @@
+class Kamal::Cli::App::SslCertificates
+  attr_reader :host, :role, :sshkit
+  delegate :execute, :info, to: :sshkit
+
+  def initialize(host, role, sshkit)
+    @host = host
+    @role = role
+    @sshkit = sshkit
+  end
+
+  def run
+    if role.running_proxy? && role.proxy.custom_ssl_certificate?
+      info "Writing SSL certificates for #{role.name} on #{host}"
+      execute *app.create_ssl_directory
+      if cert_content = role.proxy.certificate_pem_content
```

Comment:
I think we should avoid including the contents of the cert and pem files in the shell command as they could be leaked in the process lists.

So I think we want to:

1. Create a temporary file locally containing the cert with permissions 600
2. Upload the file via SSHKit (also with permissions 600)

Reply:
Oh actually no need for the tempfile, you can pass a StringIO directly to upload!